src/python/pants/backend/python/rules/pex.py

```diff
@@ -181,12 +187,17 @@ class CreatePex:
 
 
 @dataclass(frozen=True)
-class Pex(HermeticPex):
+class Pex:
     """Wrapper for a digest containing a pex file created with some filename."""
 
     directory_digest: Digest
     output_filename: str
 
+    def create_hermetic_pex_request(self, exe_req: ExecuteProcessRequest) -> HermeticPexRequest:
+        if not exe_req.input_files:
+            exe_req = dataclasses.replace(exe_req, input_files=self.directory_digest,)
+        return HermeticPexRequest(pex_path=self.output_filename, exe_req=exe_req)
```

Comment:
Oh, hm, the factory method I proposed above on `HermeticPexRequest` should actually probably move here, since this is what downstream rules will end up calling.